sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala

```diff
@@ -47,8 +47,7 @@ trait CodegenInterpretedPlanTest extends PlanTest {
     super.test(testName + " (codegen path)", testTags: _*)(
       withSQLConf(SQLConf.CODEGEN_FACTORY_MODE.key -> codegenMode) { testFun })(pos)
     super.test(testName + " (interpreted path)", testTags: _*)(
-      withSQLConf(SQLConf.CODEGEN_FACTORY_MODE.key -> interpretedMode) {
-        withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false") { testFun }})(pos)
+      withSQLConf(SQLConf.CODEGEN_FACTORY_MODE.key -> interpretedMode) {{ testFun }})(pos)
```

Comment:
Sorry, I didn't get why you reverted https://github.com/apache/spark/pull/41467 , and why do you need double curly  brackets.

Reply:
> Sorry, I didn't get why you reverted https://github.com/apache/spark/pull/41467

If when we support set `spark.sql.codegen.factoryMode` to `NO_CODEGEN` then disable WholeStageCodegen in this PR(just like set WHOLESTAGE_CODEGEN_ENABLED to false). The change seem unnecessary. This idea come from I collect other codegen config mistakes. The `NO_CODEGEN` behavior not right at now.

> why do you need double curly brackets.

My mistake, I will fix it. Thanks